scio-core/src/main/scala/com/spotify/scio/values/BloomFilter.scala

```diff
@@ -0,0 +1,127 @@
+package com.spotify.scio.values
+
+import java.io.{ByteArrayOutputStream, InputStream, OutputStream}
+
+import com.google.common.hash.{Funnel, BloomFilter => gBloomFilter}
+import com.spotify.scio.coders.Coder
+import org.apache.beam.sdk.coders.AtomicCoder
+
+/**
+ * An immutable wrapper over Guava's Bloom Filter.
+ *
+ * Serializable as a field in Avro Schema.
+ *
+ * @see companion object [[BloomFilter]]
+ */
+case class BloomFilter[T] private (private val internal: gBloomFilter[T]
+) extends ApproxFilter[T] {
+
+  def expectedFpp: Double = internal.expectedFpp()
```

Comment:
Is `internal.expectedFpp` expensive, like does it do complex math? Since the BF is immutable this can be a `val`, and part of parent trait?